src/Rendering/Graph.hs

```diff
@@ -75,23 +79,39 @@ instance (ConstructorName syntax, Foldable syntax) =>
     termAlgebra (In ann syntax) = do
       i <- fresh
       parent <- ask
-      let root = vertex $ TermVertex (fromIntegral i) (T.pack (constructorName syntax)) (converting #? Loc.span ann)
+      let root = vertex $ defMessage
+            & P.vertexId .~ fromIntegral i
+            & P.term .~ T.pack (constructorName syntax)
+            & P.maybe'span .~ (converting #? Loc.span ann)
       subGraph <- foldl' (\acc x -> overlay <$> acc <*> local (const root) x) (pure mempty) syntax
       pure (parent `connect` root `overlay` subGraph)
 
 instance (ConstructorName syntax, Foldable syntax) =>
   ToTreeGraph DiffTreeVertex (DiffF syntax Loc Loc) where
   toTreeGraph d = case d of
-    Merge t@(In (a1, a2) syntax)     -> diffAlgebra t  (Merged   (Just (MergedTerm (T.pack (constructorName syntax)) (ann a1) (ann a2))))
-    Patch (Delete t1@(In a1 syntax)) -> diffAlgebra t1 (Deleted  (Just (DeletedTerm (T.pack (constructorName syntax)) (ann a1))))
-    Patch (Insert t2@(In a2 syntax)) -> diffAlgebra t2 (Inserted (Just (InsertedTerm (T.pack (constructorName syntax)) (ann a2))))
+    Merge t@(In (a1, a2) syntax)     -> diffAlgebra t . DiffTreeVertex'Merged $ defMessage
```

Comment:
I’m having a lot of trouble understanding the control flow here, in part due to the combined use of `$` and `&`. Can you check my understanding, please? To wit: we’re starting with the default message, setting a bunch of fields on it with the `& …` lines below, and finally wrapping it up in the `DiffTreeVertex'Merged` constructor and passing that to the algebra. Is that right?

Reply:
If so, I’m not sure I really love that idiom relative to using constructors directly, but perhaps that’s just not the way this approach works…?

Reply:
Yeah, it’s… it’s not the world’s most natural idiom in Haskell. But avoiding constructors and updating a default value through lenses lets you hide implementation details such as the `_unknownFields` datum attached to message types, which gracefully handles the presence of fields we don’t recognize or can’t decrypt. (`proto3-suite` just drops that information on the ground silently). 

Remember that in protobuf records, every type is, essentially, an instance of `Lower`—given a message, every message field inside it is a `Maybe` defaulting to `Nothing`, every string field defaults to “”, and every numeric field defaults to 0. This means there’s always a sensible `defMessage` (similar in essence to `lowerBound`).

Reply:
Yeah, excellent points all around. I wouldn’t mind if we could use a (smart) constructor (ideally generated) that would include all the fields except for `_unknownFields`, tho, e.g.:

```haskell
thing :: Foo -> Bar -> Thing
thing foo bar = defMessage
  & P.foo .~ foo
  & P.bar .~ bar
```

Guessing I should file that upstream?

> Remember that in protobuf records, every type is, essentially, an instance of `Lower`

This is _such_ a good way of describing this ☝️ 👏

Reply:
Upon reflection, it occurs to me that it’s probably a good thing that the idiom for the `proto-lens` code is substantially different than that for our typical internal types. Tho it does kind of reinforce my suspicion that this is a hint that this code should be working on internal types if possible which we bridge to wire types. (Tho that would involve significant restructuring to handle the JSON formatting of the wire types, so maybe I should just shush and let you get on with things.)

Either way, I think any of the above that we decide to tackle can wait for future PRs. This is great work, @tclem.